Clear DumpRenderTree and WebKitTestRunner defaults before running tests

smfr opened the pull request:
#### 165e0db5c7299e734a4a6c04f157a835d3ee7452
<pre>
Clear DumpRenderTree and WebKitTestRunner defaults before running tests
<a href="https://bugs.webkit.org/show_bug.cgi?id=251397">https://bugs.webkit.org/show_bug.cgi?id=251397</a>
rdar://104838472

Reviewed by Jonathan Bedard.

Historically DumpRenderTree had no bundleID specified, so would save NSUserDefaults using
&quot;DumpRenderTree&quot;. WebKitTestRunner, on the other hand, was correctly set up to use
&quot;com.apple.WebKit.WebKitTestRunner&quot;. This patch gives DumpRenderTree a bundleID with
a &quot;com.apple.WebKit&quot; prefix.

These bundleIDs matter because webkitpy uses them to clear NSUserDefaults before test runs;
this worked fine for DumpRenderTree, but failed for WebKitTestRunner because the bundleID
was incorrect. Fix that in mac.py, and use the new bundleID com.apple.WebKit.DumpRenderTree too.

Generally failing to clear NSUserDefaults before testing isn&apos;t an issue, but defaults like WebCoreLogging
can get saved into NSUserDefaults if you run WebKitTestRunner directly with logging arguments, so clearing
before running tests is useful in that scenario.

* Tools/DumpRenderTree/mac/Configurations/DumpRenderTree.xcconfig:
* Tools/DumpRenderTree/mac/Info.plist:
* Tools/Scripts/webkitpy/port/mac.py:
(MacPort.reset_preferences):

Canonical link: <a href="https://commits.webkit.org/259687@main">https://commits.webkit.org/259687@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/6c884c3943bf6b99872620d100598fe7d662ef40

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/105593 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/14679 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/38476 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/114831 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/174978 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/109493 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/16091 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/5892 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/97874 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/114659 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/111347 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/12233 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/95215 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/39726 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/5/builds/109034 "Passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/94097 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/26852 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/81432 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/7957 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/28207 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/8453 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/4796 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/14079 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/47751 "Passed tests") | | 
| [❌ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/6694 "Failed to push commit to Webkit repository") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/10005 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/be87ac364f45d1c8290eddacf13f274857497215. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/9348

aproskuryakov commented:
Is this observable / verifiable? I vaguely remember seeing defaults actually being in `DumpRenderTree` domain in the past, but I don't have any on my machine to confirm.

smfr commented:
I think defaults would have been in the `DumpRenderTree` domain because we never set a bundleID. I verified that setting the bundleID in the project puts the defaults in the `com.apple.WebKit.DumpRenderTree` domain.

emw-apple commented on Tools/DumpRenderTree/DumpRenderTree.xcodeproj/project.pbxproj:
```diff
@@ -1180,13 +1180,15 @@
 			isa = XCBuildConfiguration;
 			baseConfigurationReference = BCB281F00CFA713D007E533E /* DumpRenderTree.xcconfig */;
 			buildSettings = {
+				PRODUCT_BUNDLE_IDENTIFIER = com.apple.WebKit.DumpRenderTree;
```
Could you make this change in `Tools/DumpRenderTree/mac/Configurations/DumpRenderTree.xcconfig` instead?

smfr commented:
> So to confirm, everything works fine with or without the proposed change, and it's just cleanup? Or is something broken?

Everything works fine for most people. If, like me, you run WebKitTestRunner directly with the `--webcore-logging` argument that sets the log channel NSUserDefault, then subsequent `run-webkit-test` invocations run with the logging enabled, causing lots of spew.

Since we have logic to reset DRT and WTR NSUserDefaults, which presumably worked in the past, I'm fixing it to work again.

aproskuryakov commented:
I think that I understand now. It's just that most of the PR is not about fixing the issue.

When one runs `DumpRenderTree` with `--webcore-logging`, the default gets written to `DumpRenderTree` domain, which is the domain that's currently being cleared by `run-webkit-tests`. So there is no bug with `DumpRenderTree`.

WebKitTestRunner on the other hand saves defaults to `com.apple.WebKit.WebKitTestRunner` domain, so this part is needed to fix the problem, and the rest is not:

```
-        for domain in ["DumpRenderTree", "WebKitTestRunner"]:
+       for domain in ["DumpRenderTree", "com.apple.WebKit.WebKitTestRunner"]:
```

I don't object to landing this as is, but please update the commit message to explain what part is fixing a problem, and what part is a drive-by consistency improvement. Please also confirm that nothing bad happens to DumpRenderTreeApp in iOS.

aproskuryakov commented:
Also, great catch with regards to `WebKitTestRunner` defaults!

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/6c884c3943bf6b99872620d100598fe7d662ef40. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/9348

smfr committed:
Clear DumpRenderTree and WebKitTestRunner defaults before running tests
https://bugs.webkit.org/show_bug.cgi?id=251397
rdar://104838472

Reviewed by Jonathan Bedard.

Historically DumpRenderTree had no bundleID specified, so would save NSUserDefaults using
"DumpRenderTree". WebKitTestRunner, on the other hand, was correctly set up to use
"com.apple.WebKit.WebKitTestRunner". This patch gives DumpRenderTree a bundleID with
a "com.apple.WebKit" prefix.

These bundleIDs matter because webkitpy uses them to clear NSUserDefaults before test runs;
this worked fine for DumpRenderTree, but failed for WebKitTestRunner because the bundleID
was incorrect. Fix that in mac.py, and use the new bundleID com.apple.WebKit.DumpRenderTree too.

Generally failing to clear NSUserDefaults before testing isn't an issue, but defaults like WebCoreLogging
can get saved into NSUserDefaults if you run WebKitTestRunner directly with logging arguments, so clearing
before running tests is useful in that scenario.

* Tools/DumpRenderTree/mac/Configurations/DumpRenderTree.xcconfig:
* Tools/DumpRenderTree/mac/Info.plist:
* Tools/Scripts/webkitpy/port/mac.py:
(MacPort.reset_preferences):

Canonical link: https://commits.webkit.org/259687@main

webkit-commit-queue commented:
Committed 259687@main (165e0db5c729): <https://commits.webkit.org/259687@main>

Reviewed commits have been landed. Closing PR #9348 and removing active labels.
